packages/autocomplete-core/src/__tests__/getInputProps.test.ts

```diff
@@ -1112,23 +1112,405 @@ describe('getInputProps', () => {
+          };
+          const { inputElement } = createPlayground(createAutocomplete, {
+            navigator,
+            defaultActiveItemId: 0,
+            initialState: {
+              isOpen: true,
+              collections: [
+                createCollection({
+                  source: { onSelect },
+                  items: [
+                    { label: '1', url: '#1' },
+                    { label: '2', url: '#2' },
+                  ],
+                }),
+              ],
+            },
+          });
+
+          inputElement.focus();
+          userEvent.type(inputElement, '{ctrl}{enter}');
+
+          expect(navigator.navigateNewTab).toHaveBeenCalledTimes(0);
+        });
+
+        test('calls onSelect with item URL', () => {
+          const onSelect = jest.fn();
+          const navigator = {
+            navigate: jest.fn(),
+            navigateNewTab: jest.fn(),
+            navigateNewWindow: jest.fn(),
+          };
+          const {
+            inputElement,
+            refresh,
+            setCollections,
+            setContext,
+            setIsOpen,
+            setQuery,
+            setActiveItemId,
+            setStatus,
+          } = createPlayground(createAutocomplete, {
+            navigator,
+            defaultActiveItemId: 0,
+            initialState: {
+              isOpen: true,
+              collections: [
+                createCollection({
+                  source: { onSelect, getItemUrl: ({ item }) => item.url },
+                  items: [
+                    { label: '1', url: '#1' },
+                    { label: '2', url: '#2' },
+                  ],
+                }),
+              ],
+            },
+          });
+
+          inputElement.focus();
+          userEvent.type(inputElement, '{ctrl}{enter}');
+
+          expect(onSelect).toHaveBeenCalledTimes(1);
+          expect(onSelect).toHaveBeenCalledWith({
+            event: expect.any(KeyboardEvent),
+            item: {
+              label: '1',
+              url: '#1',
+            },
+            itemInputValue: '',
+            itemUrl: '#1',
+            refresh,
+            setCollections,
+            setContext,
+            setIsOpen,
+            setQuery,
+            setActiveItemId,
+            setStatus,
+            source: expect.any(Object),
+            state: {
+              collections: [
+                {
+                  items: [
+                    {
+                      label: '1',
+                      url: '#1',
+                    },
+                    {
+                      label: '2',
+                      url: '#2',
+                    },
+                  ],
+                  source: expect.any(Object),
+                },
+              ],
+              completion: null,
+              context: {},
+              isOpen: false,
+              query: '',
+              activeItemId: 0,
+              status: 'idle',
+            },
```

Comment:
That creates a lot of noise. We don't care too much about the state value so that's fine IMO:

```suggestion
            state: expect.any(Object),
```

Reply:
Done, but if we do not assert the state value then I do not find this assertion very useful.

Maybe remove the assertion of the arguments and just keep the one that checks that the function was called?

Reply:
The check is still useful for `item` and `itemUrl`.